[Terng Yan Long] iP

Korochii opened the pull request:
# HoloBot

> "Fun things are fun" -  Hirasawa Yui

HoloBot frees your mind of having to remember things you need to do. It's,

- GUI-Based
- Easy to learn AND use!

All you need to do is,

1. Clone this [repository](https://github.com/Korochii/ip)
2. Run `Launcher.java` using your favourite editor/terminal
3. And play with the different commands available :smiley:

AND of course, it is **FREE**!

Features:
- [x] Managing different type of tasks

You can customize the images used in the bot as well!
Simply edit `MainWindow.java` and you're good to go:
``` java
private Image userImage = new Image(this.getClass().getResourceAsStream("/images/Rushia.png"));
private Image dukeImage = new Image(this.getClass().getResourceAsStream("/images/Pekora.png"));
```

Commit:
Add Gradle support

Korochii committed:
Implement greet/echo/exit function

Korochii committed:
Remove class files

Korochii committed:
Fix exclusion of class files in gitignore

Korochii committed:
Fix exclusion of class files in gitignore

Korochii committed:
Finalize submission for Level-1

Korochii committed:
Finalize submission for Level-1

Korochii committed:
Merge branch 'master' of github.com:Korochii/ip

Korochii committed:
Add javadoc for echo function

Korochii committed:
Update javadoc for echo function

Korochii committed:
Remove extra comments at the end of the file

Korochii committed:
Implement Level-2 (add/list)

Korochii committed:
Adjust formatting of identation for display function

Korochii committed:
Represent tasks using a separate class

Korochii committed:
Update javadocs for the Task class

Korochii committed:
Include checkboxfor each task when displaying list

Korochii committed:
Implement mark and unmark functions

Korochii committed:
Implement deadline/event/todo as extension to task

Korochii committed:
Implement Level-4 (ToDos, Events, Deadlines)

Korochii committed:
Include testcases for functions that have been implemented

Korochii committed:
Implement Level-5 using DukeExceptions

Korochii committed:
Use ArrayList to store tasks

Korochii committed:
Implement Level-6 (Delete)

Korochii committed:
Implement Level-6 (Delete)

Korochii committed:
Implement read saved data feature when Duke starts up

Korochii committed:
Implement Level-7 (Save)

Korochii committed:
Edit gitignore to exclude data files

Korochii committed:
Implement Level-8 (Date/Time)

Korochii committed:
Implement Level-8 (Date/Time)

Korochii committed:
Merge from branch-Level-8

Korochii committed:
Remove data folder

Korochii committed:
Fix parsing function in loadData()

Korochii committed:
Fix parsing function in loadData()

Korochii committed:
Extract code related to Ui into another class

Korochii committed:
Extract code related to Parser into another class

Korochii committed:
Remove redundant variables in Duke.java

Korochii committed:
Refactor code into packages

Korochii committed:
Refactor code into packages

Korochii committed:
Implement JUnit Tests

Korochii committed:
Add invalid command testcase to ParserTest

Korochii committed:
Remove data file from src/test folder

Korochii committed:
Add JavaDoc comments to the code

Korochii committed:
Ensure coding standard is being followed

Korochii committed:
Implement Level-9 (find)

Korochii committed:
Merge branch 'A-CodingStandard'

Korochii committed:
Merge branch 'branch-Level-9'

Korochii committed:
Fix typo in parser (find)

Korochii committed:
Update version number in Duke

Korochii committed:
Merge branch 'master' into add-gradle-support

Korochii committed:
Add gradle support toDuke

Korochii committed:
Fix checkstyle errors in code

Korochii committed:
Implement basic features of JavaFX

Korochii committed:
Edit return string in TaskList.find()

Korochii committed:
Implement Level-10 (GUI)

Kidsnd274 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,40 @@
+package duke;
+
+import duke.exception.DukeException;
+import duke.misc.Parser;
```
Oh nice, I probably should segment my code into smaller packages as well.

Kidsnd274 commented on src/main/java/duke/exception/DukeException.java:
```diff
@@ -0,0 +1,20 @@
+ */
+public class DukeException extends RuntimeException {
+    /**
+     * Initialises a new instance of DukeException.
```
I think the term for creating a new instance of a class is instantiate instead of initialize.

Kidsnd274 commented on src/main/java/duke/exception/InvalidCommand.java:
```diff
@@ -0,0 +1,30 @@
+     */
+    @Override
+    public String toString() {
+        return "OOPS!!! " + this.errorMessage;
```
I'm really nitpicking here because you've pretty much made like 0 mistakes, but according to a friend, we aren't supposed to use the "this" keyword if possible unless absolutely necessary. Don't know if it's true or not, but I also did use "this" everywhere in my code.

Kidsnd274 commented on src/main/java/duke/misc/Parser.java:
```diff
@@ -0,0 +1,271 @@
+        List<String> wordsList = Arrays.asList(wordsArray);
+        String firstWord = wordsList.get(0);
+
+        switch (firstWord) {
```
Nice, I like this neatness of this code. I used an if-else because switch-case had issues I couldn't solve on my end but seems like the switch-case statements really makes it a lot neater looking.

Kidsnd274 commented on src/test/java/duke/task/DeadlineTest.java:
```diff
@@ -0,0 +1,67 @@
+
+import org.junit.jupiter.api.Test;
+
+public class DeadlineTest {
```
Nice test cases!

Kidsnd274 reviewed (commented):
Nice job on this project! Pretty much nothing to comment on, so I was really scraping the barrel trying to come up with comment haha.

sophiien commented on src/main/java/duke/misc/Storage.java:
```diff
@@ -0,0 +1,217 @@
+     * @param f File that contains the data.
+     * @param listOfTasks Target TaskList to store the data.
+     */
+    public static void loadData(File f, TaskList listOfTasks) {
```
Rather long method, maybe can try to abstract some functionalities like the splitting of strings?

sophiien commented on src/main/java/duke/misc/Storage.java:
```diff
@@ -0,0 +1,217 @@
+                    listOfTasks.getListOfTasks().add(currTask);
+                    listOfTasks.incrementTasks();
+                } else if (currCommand.startsWith("[E]")
+                        & currCommand.contains("(at:")
```
Indentation for next line should be 8 spaces

sophiien reviewed (commented):
Overall code looks neat. Added some suggestions good job :)

Korochii committed:
Address feedback by peers + Fix GUI text wrapping

Korochii committed:
Implement A-Assertions

Use assert statement in Storage:initTaskList to check if the user-specified size is valid (must be more than 0), before creating a new TaskList.

Korochii committed:
Implement A-CodeQuality
Current code is not clean enough and can be improved further.

In Parser.java and Storage.java, there are duplicate codes as well as
excessively long methods.

Let's create new methods that abstracts the creation of task, such
that Storage:loadData and Parser:parse will not be as long and will
be clearer. We can also create a new class that deals with checking
of date and time, which allows us to abstract the relevant methods
away from Parser and Storage.

In addition, we can also remove the Ui class and Storage:readData
since we have shifted to using GUI.

Korochii committed:
Merge pull request #2 from Korochii/branch-A-Assertions

Implement A-Assertions

Korochii committed:
Merge branch 'master' into branch-A-CodeQuality

Korochii committed:
Merge pull request #3 from Korochii/branch-A-CodeQuality

Implement A-CodeQuality

Korochii committed:
Set up CI

Korochii committed:
Implement the schedule function (BCD-Extensions)

Korochii committed:
Merge pull request #4 from Korochii/b-B-ViewSchedules

Implement the schedule function (BCD-Extensions)

Korochii committed:
Fix CI

Korochii committed:
Change image of duke/user

Korochii committed:
Remove unused images

Korochii committed:
Change background of GUI

Korochii committed:
Fix background issue when scrolling

Korochii committed:
Change formatting of text-box in GUI

Korochii committed:
Add personality to the bot

Korochii committed:
Add product screenshot

Korochii committed:
Set theme jekyll-theme-cayman

Korochii committed:
Delete _config.yml

Korochii committed:
Update README.md

Korochii committed:
Update README.md

Korochii committed:
Add pictures for user guide

Korochii committed:
Update README.md

Korochii committed:
Edit archiveBaseName in build.gradle

Korochii committed:
Edit mainClassName in build.gradle

Korochii committed:
Edit build.gradle

Korochii committed:
Add font file to support cross-platform

Korochii committed:
Fix checkstyle errors

Korochii committed:
Refactor Code

Korochii committed:
Update README.md
